src/bin/load_test.rs

```diff
@@ -199,39 +236,170 @@ async fn main() {
+
+            let mut data_contract_variant = data_contract.clone();
+            data_contract_variant.set_id(new_id);
+            CreatedDataContract::from_contract_and_entropy(
+                data_contract_variant,
+                entropy.into(),
+                PlatformVersion::latest(),
+            )
+            .expect("expected to get contract")
+        })
+        .collect::<Vec<_>>();
+
+    let partial_identity = identity.clone().into_partial_identity_info();
+
+    let key_to_use = identity
+        .get_first_public_key_matching(
+            Purpose::AUTHENTICATION,
+            HashSet::from([SecurityLevel::CRITICAL]),
+            HashSet::from([KeyType::ECDSA_SECP256K1]),
+        )
+        .expect("expected to get a key");
+
+    let mut transitions_queue = VecDeque::new();
+
+    let first_exists = DataContract::fetch(
+        &sdk,
+        data_contract_variants.first().unwrap().data_contract().id(),
+    )
+    .await
+    .expect("expected to get data contract")
+    .is_some();
+
+    for data_contract_variant in data_contract_variants.iter().rev() {
+        let exists = if !first_exists {
+            false
+        } else {
+            DataContract::fetch(&sdk, data_contract_variant.data_contract().id())
+                .await
+                .expect("expected to get data contract")
+                .is_some()
+        };
+        if !exists {
+            let transition = DataContractCreateTransition::new_from_data_contract(
+                data_contract_variant.data_contract().clone(),
+                *data_contract_variant.entropy_used(),
+                &partial_identity,
+                key_to_use.id(),
+                signer.as_ref(),
+                PlatformVersion::latest(),
+                None,
+            )
+            .expect("expected transition");
+
+            transitions_queue.push_front(transition); // we push to the front on purpose
+        } else {
+            break; // if one exists we can assume all previous ones also exist
+        }
+    }
+
+    for (i, transaction) in transitions_queue.iter().enumerate() {
+        tracing::info!("registering contract {}", i);
+        if i % 24 == 0 {
+            transaction
+                .broadcast_and_wait(&sdk, None)
+                .await
+                .expect("expected to register contract");
+        } else {
+            transaction
+                .broadcast(&sdk)
+                .await
+                .expect("expected to register contract");
+        }
+    }
+
+    data_contract_variants
+        .into_iter()
+        .map(|c| c.data_contract_owned())
+        .collect()
+}
+
 async fn broadcast_random_documents_load_test(
     sdk: Arc<Sdk>,
     identity: &Identity,
-    identity_private_keys: &IdentityPrivateKeysMap,
-    data_contract: Arc<DataContract>,
-    document_type: Arc<DocumentType>,
+    signer: Arc<SimpleSigner>,
+    document_type: DocumentType,
+    document_type_variants: Vec<Arc<DocumentType>>,
```

Comment:
We LOVE `Arc`s! :)

Reply:
Arc is actually right in this situation, but not all situations where it has been used.